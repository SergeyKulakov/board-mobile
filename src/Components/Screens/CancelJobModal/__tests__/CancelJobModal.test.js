import React from 'react'
import { shallow } from 'enzyme'

import 'jest-styled-components'

import Component from '../CancelJobModal'

describe('CancelJobModal', () => {
  const defaultProps = {
    navigate: {
      push: () => null,
      pop: () => null,
      hideModal: () => null,
    },
    onSubmit: () => null,
    t: () => 'text',
  }

  it('render snapshot', () => {
    const wrapper = shallow(<Component {...defaultProps} />)

    expect(wrapper).toMatchSnapshot()
  })

  it('submit action', () => {
    const props = {
      ...defaultProps,
      onSubmit: jest.fn(),
      navigate: {
        ...defaultProps.navigate,
        hideModal: jest.fn(),
      },
    }

    const wrapper = shallow(<Component {...props} />)

    wrapper.setState({ reason: 'reason' })

    wrapper
      .find('Button')
      .props()
      .onClick()

    expect(props.onSubmit).toHaveBeenCalledTimes(1)
    expect(props.onSubmit).toHaveBeenCalledWith('reason')
    expect(props.navigate.hideModal).toHaveBeenCalledTimes(1)
  })
})
