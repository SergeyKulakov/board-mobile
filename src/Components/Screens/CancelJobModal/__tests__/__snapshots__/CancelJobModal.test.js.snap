// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CancelJobModal render snapshot 1`] = `
<Styled(View)>
  <Header
    onBackClick={[Function]}
  />
  <Styled(ScrollViewMock)>
    <Styled(TextInput)
      onChangeText={[Function]}
      value=""
    />
  </Styled(ScrollViewMock)>
  <Button
    linear={true}
    onClick={[Function]}
    text="text"
  />
</Styled(View)>
`;
